radix-engine-common/src/math/decimal.rs

```diff
@@ -181,6 +181,35 @@ impl Decimal {
                     }
                 }
             }
+            RoundingMode::MidpointNearestEven => {
+                if self.0 % divisor == BnumI256::zero() {
+                    self.clone()
+                } else {
```

Comment:
This all feels like quite an expensive calculation, but I can't see any obvious speed-ups (although I imagine there would be some if we wanted to do a separate optimisation pass in a future workstream).

I'm glad that we're outsourcing this to engine where we can, because I imagine this would be quite gas intensive.

Reply:
In the fixed version of implementation, `remainder` will be checked against `mid_point`, so it's not a waste, assuming the `eq` check is relatively cheap. We can benchmark more later on for opitmisation.